source/macos/AdaptiveCards/AdaptiveCards/Views/ACRCollectionViewItem.swift

```diff
@@ -2,37 +2,41 @@ import AdaptiveCards_bridge
 import Cocoa
 
 class ACRCollectionViewItem: NSCollectionViewItem {
-    static let identifier = NSUserInterfaceItemIdentifier(rawValue: "MyItem")
+    static let identifier = NSUserInterfaceItemIdentifier(rawValue: "ImageSetItem")
     
     override func loadView() {
-        view = itemImageView
+        view = itemView
     }
     
-    private let itemImageView: NSImageView = {
-        let view = NSImageView()
+    private let itemView: NSView = {
+        let view = NSView()
         view.translatesAutoresizingMaskIntoConstraints = false
-        view.imageAlignment = .alignCenter
         return view
     }()
     
-    func setupItem(with urlString: String, hostConfig: ACSHostConfig, imageSize: ACSImageSize) {
-        guard let url = URL(string: urlString) else {
-            logError("URL is invalid")
-            return
-        }
-        DispatchQueue.global().async {
-            guard let data = try? Data(contentsOf: url) else { return }
-            DispatchQueue.main.async {
-                guard let image = NSImage(data: data) else { return }
-                // Add
-                let imageRatio = ImageUtils.getAspectRatio(from: image.size)
-                var maxImageSize = ImageUtils.getImageSizeAsCGSize(imageSize: imageSize, width: 0, height: 0, with: hostConfig, explicitDimensions: false)
-                if imageRatio.height < 1 {
-                    maxImageSize.height *= imageRatio.height
-                }
-                image.size = maxImageSize
-                self.itemImageView.image = image
-            }
-        }
+//    func setupItem(with urlString: String, hostConfig: ACSHostConfig, imageSize: ACSImageSize) {
+//        guard let url = URL(string: urlString) else {
+//            logError("URL is invalid")
+//            return
+//        }
+//        DispatchQueue.global().async {
+//            guard let data = try? Data(contentsOf: url) else { return }
+//            DispatchQueue.main.async {
+//                guard let image = NSImage(data: data) else { return }
+//                // Add
+//                let imageRatio = ImageUtils.getAspectRatio(from: image.size)
+//                var maxImageSize = ImageUtils.getImageSizeAsCGSize(imageSize: imageSize, width: 0, height: 0, with: hostConfig, explicitDimensions: false)
+//                if imageRatio.height < 1 {
+//                    maxImageSize.height *= imageRatio.height
+//                }
+//                image.size = maxImageSize
+//                self.itemImageView.image = image
+//            }
+//        }
+//    }
+    
+    func setupBounds(with imageView: ImageSetImageView, and imageSize: ACSImageSize, hostConfig: ACSHostConfig) {
+        view.addSubview(imageView)
+        view.invalidateIntrinsicContentSize()
```

Comment:
```suggestion
        imageView.removeFromSuperview()
        view.addSubview(imageView)
        imageView.frame = view.bounds
```

Reply:
Done
Updating the image size now as part of the `setImage` protocol method